src/helpers/crypto.ts

```diff
@@ -1,15 +1,75 @@
 import * as nacl from "tweetnacl";
 import { HexString } from "@frequency-chain/ethereum-utils";
 import { u8aToHex } from "@polkadot/util";
+import {
+  SiwfResponseCredentialGraph,
+  SiwfResponseCredentialRecoverySecret,
+  VerifiedRecoverySecret,
+} from "@projectlibertylabs/siwf";
+import { multibaseSr25519 } from "./utils";
 
-export function generateGraphKeyPair(): {
-  privateKey: HexString;
-  publicKey: HexString;
+export function generateGraphKeyCredential(
+  keyPair: nacl.BoxKeyPair,
+): SiwfResponseCredentialGraph {
+  const issuer = multibaseSr25519(keyPair.publicKey);
+  return {
+    "@context": [
+      "https://www.w3.org/ns/credentials/v2",
+      "https://www.w3.org/ns/credentials/undefined-terms/v2",
+    ],
+    type: ["VerifiedGraphKeyCredential", "VerifiableCredential"],
+    issuer: `did:key:${issuer}`,
+    validFrom: new Date().toISOString(),
+    credentialSchema: {
+      type: "JsonSchema",
+      id: "https://schemas.frequencyaccess.com/VerifiedGraphKeyCredential/bciqmdvmxd54zve5kifycgsdtoahs5ecf4hal2ts3eexkgocyc5oca2y.json",
+    },
+    credentialSubject: {
+      id: `did:key:${issuer}`,
+      encodedPublicKeyValue: keyPair.publicKey.toString(),
```

Comment:
@wilwade confirming that the `issuer` and `id` values are correct

Reply:
Side note: you will want to make sure that `keyPair.publicKey.toString()` has a `0x` prefix for the `encodedPublicKeyValue`.

Because these are from the control key and it is secp256k1, the `did:key` should be the multibase of the secp256k1:

```ts
export function multibaseSecp256k1(address: string | Uint8Array): string {
  const keyBuf =
    typeof address === "string"
      ? Buffer.from(address.replace("0x", ""), "hex")
      : Buffer.from(address);
  return "z" + bs58.encode(Buffer.concat([Buffer.from("0701", "hex"), keyBuf]));
}
```

Per: https://w3c-ccg.github.io/did-key-spec/#signature-method-creation-algorithm

@aramikm we should also give examples of these on siwf docs as well

Reply:
And the issuer should be the ethereum control key address, not the public key of the pair

Reply:
That's helpful but the actual question I raised was whether the issuer in both the credential context and the credential subject are supposed to be the same.  In the example for an email credential, the issuer in the context is "did:web:frequencyaccess.com"

Reply:
Update from a conversation with @aramikm and myself.

There isn't a multibase encoding (even in draft) for Ethereum Address (there is for the secp256k1 public key, but we don't have that, we only have the Ethereum Address).

What we can use however, is `did:ethr` https://github.com/decentralized-identity/ethr-did-resolver

This is ok to use per the `did:ethr` spec as it doesn't require any chain registration before use.

So @claireclark1 you just need to update from `did:key` to `did:ethr` and make sure that the `accountId` is the ethereum address with the `0x` prefix (I think it is and in that case the only change is `did:key` -> `did:ethr`).

The same would apply to the `generateRecoverySecretCredential`